### PR TITLE
[Feature][Branch][Snackbar] Action 

### PR DESCRIPTION
**Pull request by Ryz0nd**
## Description
default_snackbar1
with_action_snackbar1

- default click behavior might close snackbars

- It will be at right side of snackbar when short

- It will be at the bottom of it when long

- It allows users to choose short or long(text or timer).

- Action function is separated into another story

## Related Issues

#46 

## Tests

I added the following tests:

 - Just updated snapshot.

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui-native/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/components/shared/Snackbar/index.tsx [hunk 1]
```diff
@@ -1,9 +1,17 @@
 import * as React from 'react';
 
-import { Text, TextStyle, ViewStyle } from 'react-native';
+import { Text, TextStyle, ViewStyle, Dimensions } from 'react-native';
```
Comment (JongtaekChoi) [inline src/components/shared/Snackbar/index.tsx]: 'Dimensions' must be in the first position to keep the import sorting rule.
It makes the CircleCI test failed. 
If your code didn't pass the CircleCI tests and you want to what is the reason just click the 'Details' button. 
Then you can find the reason. 
image

File: src/components/shared/Snackbar/index.tsx [hunk 2]
```diff
@@ -13,12 +21,20 @@ const Container = styled.View`
...
+  padding-left: 16px;
+`;
+
 export interface SnackbarProps {
   text: string;
   testID?: string;
   show: boolean;
   setShow: (show: boolean) => void;
   timer?: Timer;
+  action?: React.ReactNode;
```
Comment (JongtaekChoi) [inline src/components/shared/Snackbar/index.tsx]: I think only text, text color, and callback should be provided as action. 
Using ReactNode can burden users with implementation and cause break the struct.
How do you think?

Comment (Ryz0nd) [reply src/components/shared/Snackbar/index.tsx]: That would be better.
Thank you for a good idea.
I'll fix it right away.

File: src/components/shared/Snackbar/index.tsx (hunk 1 above)
Comment (Ryz0nd) [reply src/components/shared/Snackbar/index.tsx]: I'm learning a lot thanks to you!

**Review by JongtaekChoi — APPROVED** [review]

**Comment by taejs** [comment]
[Suggestion] 
I think the Snackbar story seems a little messy because there're too many features for a button.
How about creating a new button in Storybook or a new story for this feature?

**Comment by Ryz0nd** [comment]
> [Suggestion]
> I think the Snackbar story seems a little messy because there're too many features for a button.
> How about creating a new button in Storybook or a new story for this feature?

I'll give it a try.

**Comment by taejs** [comment]
Good changes, but I meant it would be better when each button shows each feature.
For now, buttons in story exist only for showing timer feature. 
So I recommend creating a new button for this feature or radio buttons that allow users to choose short or long.

**Comment by Ryz0nd** [comment]
> Good changes, but I meant it would be better when each button shows each feature.
> For now, buttons in story exist only for showing timer feature.
> So I recommend creating a new button for this feature or radio buttons that allow users to choose short or long.

Oh I see. I'll try to change it.

**Review by JongtaekChoi — APPROVED** [review]

**Comment by taejs** [comment]
LGTM!! awesome

**Pull request merged**